src/main/java/com/theironyard/entities/User.java

```diff
@@ -0,0 +1,87 @@
+package com.theironyard.entities;
+
+import com.theironyard.utilities.PasswordStorage;
+
+import javax.persistence.*;
+
+@Entity
+@Table(name = "users")
+public class User {
+    @Id
+    @GeneratedValue
+    private int id;
+
+    @Column(nullable = false)
+    private String username;
+
+    @Column(nullable = false)
+    private String email;
+
+    @Column
+    private String firstName;
+
+    @Column
+    private String lastName;
+
+    @Column
+    private String password;
+
+    public User() {
+    }
+
+    public User(String username, String password, String email, String firstName, String lastName) throws PasswordStorage.CannotPerformOperationException {
+        this.username = username;
+        this.password = PasswordStorage.createHash(password);
+        this.email = email;
+        this.firstName = firstName;
+        this.lastName = lastName;
+    }
+
+    public int getId() {
+        return id;
+    }
+
+    public void setId(int id) {
+        this.id = id;
+    }
+
+    public String getUsername() {
+        return username;
+    }
+
+    public void setUsername(String username) {
+        this.username = username;
+    }
+
+    public String getEmail() {
+        return email;
+    }
+
+    public void setEmail(String email) {
+        this.email = email;
+    }
+
+    public String getFirstName() {
+        return firstName;
+    }
+
+    public void setFirstName(String firstName) {
+        this.firstName = firstName;
+    }
+
+    public String getLastName() {
+        return lastName;
+    }
+
+    public void setLastName(String lastName) {
+        this.lastName = lastName;
+    }
+
+    public String getPassword() {
+        return password;
+    }
+
+    public void setPassword(String password) throws PasswordStorage.CannotPerformOperationException {
```

Comment:
If I am correct you will end up rehashing passwords.  When the user is pulled from the database it is already hashed which would mean it would try to hash it again.